src/main.js

```diff
@@ -346,11 +380,26 @@ app.on('ready', function() {
       event.preventDefault();
       switch (process.platform) {
         case 'win32':
+          if (config.minimizeToTray) {
+            mainWindow.hide();
+            mainWindow.isHidden = true;
+          }
+          else {
+            mainWindow.minimize();
+          }
+          break;
         case 'linux':
           mainWindow.minimize();
           break;
         case 'darwin':
-          mainWindow.hide();
+          if (config.minimizeToTray) {
+            mainWindow.hide();
+            app.dock.hide();
+            mainWindow.isHidden = true;
+          }
+          else {
+            mainWindow.minimize();
```

Comment:
This should be `mainWindow.hide()`. This is `close` event, so minimizing is incorrect behavior.